packages/checkout-storefront/src/sections/GuestUser/useGuestUserForm.ts

```diff
@@ -42,9 +42,14 @@ export const useGuestUserForm = ({ initialEmail }: GuestUserFormProps) => {
   const { setCheckoutUpdateState } = useCheckoutUpdateStateChange("checkoutEmailUpdate");
 
   const validationSchema = object({
+    createAccount: bool(),
     email: string().email(errorMessages.invalid).required(errorMessages.required),
-    password: string().min(8, formatMessage(passwordMessages.passwordAtLeastCharacters)),
-  });
+    password: string().when(["createAccount"], ([createAccount], field) =>
+      createAccount
+        ? field.min(8, formatMessage(passwordMessages.passwordAtLeastCharacters)).required()
+        : field
```

Comment:
Why are we not making it required when logging in?

Reply:
Sorry, I meant when we're not creating account - the password requirements should be the same for both, right?